rktl_sim/nodes/simulator

```diff
@@ -358,5 +449,244 @@ class Simulator(object):
+                for key in rospy_param:
+                    #recurse down the path, using the key as the continuation
+                    rospy_param[key] = self.get_sim_param(path + f'$/{key}')
+                return rospy_param
+
+            elif type_rospy == float or type_rospy == int or type_rospy == list:
+                return rospy.get_param(path)
+
+        raise self.InvalidRospyParam(path)
+
+
+def get_spawn_bounds(self):
+    """
+    randomly generates the spawn bounds if we specify to generate random ones
+    @return: the spawn bound array
+    """
+    # Setting up field
+    fw = self.get_sim_param("/field/width")
+    fl = self.get_sim_param("/field/length")
+    spawn_height = rospy.get_sim_param('~spawn_height', 0.06)
+    wt = rospy.get_sim_param('/field/wall_thickness')
+
+    spawn_bounds = [[-(fl / 2) + (2 * wt), (fl / 2) - (2 * wt)],
+                    [-(fw / 2) + (2 * wt), (fw / 2) - (2 * wt)],
+                    [spawn_height, spawn_height]]
+    return spawn_bounds
+
+
+def generate_car_properties(self):
+    """
+    Set the car parameters based on those provided in the rospy directory (the properties of the car itself: length, max_speed, throttle, steering, simulate effort, etc)
+    - generates random max speed and steering throw by calling other functions
+    - used to generate these properties for each car
+    Create a list for each car depending on its properties:
+        Car id
+        Car pose publisher
+        Car odom publisher
+        Car effort subscribers
+        Car command subscribers
+    Publisher:
+        match_status
+    Service:
+        Sim_reset
+        Sim_create_car
+        sim_delete_Car
+    @return the generic car properties that are created for thsi specific run in a dictionary
+    """
+    # general paramters that are the same for every single car
+    # if too constrained in car length between what we detect and what it is - might learn differently
+    # use the get_sim_param in each function case: makes the most sense
+    car_properties = {'length': rospy.get_sim_param('/cars/length'),
+                      'max_speed': self.get_sim_param("/cars/throttle/max_speed"),
+                      'steering_throw': self.get_sim_param("/cars/throttle/max_steering_throw"),
+                      'throttle_tau': rospy.get_sim_param("/cars/throttle/tau"),
+                      'steering_rate': rospy.get_sim_param("/cars/steering/rate"),
+                      'simulate_effort': (self.mode == SimulatorMode.REALISTIC)}
+    self.car_properties = car_properties
+    return car_properties
+
+
+def delete_all_cars(self):
+    car_configs = rospy.get_param('~cars', [])
+    self.reset_lock.acquire()
+    for car_config in car_configs:
+
+        if 'name' not in car_config:
+            rospy.signal_shutdown('no "name" set for car config in sim')
+        car_name = car_config['name']
+        if 'randomize_pose' in car_config:
+            init_pose = None
```

Comment:
I think this is leftover from previous implementation

Reply:
I removed it!